syntax = "proto3";

package pixanalyser;

message AiAnalyzeCnpjRequest {
  string id = 1;
  int32 origin_client_id = 2;
  string destination_key_value = 3;
  double amount = 4;
  string timestamp = 5;
  string key = 6;
  string key_type = 7;
  string institution = 8;
  string branch = 9;
  string account_number = 10;
  string account_type = 11;
  string owner_type = 12;
  string owner_name = 13;
  string cnpj = 14;
  string company_name = 15;
  string registration_date = 16;
  string status = 17;
  string status_date = 18;
  double share_capital = 19;
  string branch_type = 20;
  int32 common_transfers_client = 21;
  int32 all_transfers = 22;
}

message AiAnalyzeCnpjResponse {
  double confidence_score = 1;
  repeated string fraud_reasons = 2;
}

service AiAnalyzeCnpjService {
  rpc AiAnalyzeCnpj (AiAnalyzeCnpjRequest) returns (AiAnalyzeCnpjResponse);
}